plugins/browser-plugin-web-vitals/src/index.ts

```diff
@@ -0,0 +1,120 @@
+    };
+  }
+  window.webVitals.onCLS(addWebVitalsMeasurement('cls'));
+  window.webVitals.onFID(addWebVitalsMeasurement('fid'));
+  window.webVitals.onLCP(addWebVitalsMeasurement('lcp'));
+  window.webVitals.onINP(addWebVitalsMeasurement('inp'));
+  window.webVitals.onTTFB(addWebVitalsMeasurement('ttfb'));
+}
+
+/**
+ * Adds Web Vitals measurement events
+ *
+ * @param pluginOptions.loadWebVitalsScript - Should the plugin immediately load the Core Web Vitals measurement script from UNPKG CDN.
+ * @param pluginOptions.webVitalsSource - The URL endpoint the Web Vitals script should be loaded from.
+ * @remarks
+ */
+export function WebVitalsPlugin(pluginOptions: WebVitalsPluginOptions = defaultPluginOptions): BrowserPlugin {
+  const webVitalsObject: Record<string, unknown> = {};
+  const options = { ...defaultPluginOptions, ...pluginOptions };
+  let trackerId: string;
+  return {
+    activateBrowserPlugin: (tracker) => {
+      trackerId = tracker.id;
+      _trackers[trackerId] = tracker;
+
+      if (options.loadWebVitalsScript) {
+        const webVitalsScript = document.createElement('script');
+        webVitalsScript.setAttribute('src', options.webVitalsSource);
+        webVitalsScript.setAttribute('async', '1');
+        addEventListener(
+          webVitalsScript,
+          'error',
+          () => {
+            LOG.warn(`Failed to load ${options.webVitalsSource}`);
+          },
+          true
+        );
+
+        addEventListener(webVitalsScript, 'load', () => webVitalsListener(webVitalsObject), true);
+
+        document.head.appendChild(webVitalsScript);
+      } else {
+        webVitalsListener(webVitalsObject);
+      }
+
+      function sendValues() {
+        dispatchToTrackersInCollection(Object.keys(_trackers), _trackers, (t) => {
+          t.core.track(
+            buildSelfDescribingEvent({
+              event: {
+                schema: WEB_VITALS_SCHEMA,
+                data: webVitalsObject,
+              },
+            })
+          );
+        });
+      }
+
+      // Safari does not fire "visibilitychange" on the tab close
+      // So we have 2 options: loose Safari data, or loose LCP/CLS that depends on "visibilitychange" logic.
+      // Current solution: if LCP/CLS supported, use `onHidden` otherwise, use `pagehide` to fire the callback in the end.
+      //
+      // More details: https://github.com/treosh/web-vitals-reporter/issues/3
+      const supportedEntryTypes = (PerformanceObserver && PerformanceObserver.supportedEntryTypes) || [];
+      const isLatestVisibilityChangeSupported = supportedEntryTypes.indexOf('layout-shift') !== -1;
+
+      if (isLatestVisibilityChangeSupported) {
+        const onVisibilityChange = () => {
+          if (document.visibilityState === 'hidden') {
+            sendValues();
+            removeEventListener('visibilitychange', onVisibilityChange, true);
+          }
+        };
+        window.addEventListener('visibilitychange', onVisibilityChange, true);
```

Comment:
Bit awkward that we are using `addEventListener` from `browser-tracker-core` above but then not here. Is it just for symmetry with `removeEventListener`? For my OCD if that's the case can we specify `window.removeEventListener()` above to clarify that relationship?

Reply:
Removed all the calls to the helper, is not needed currently based on the browser targets we have.